### DISCUSSION TITLE
[While Update from 6.9 to X] Index Column size too large

### DISCUSSION THREAD

**Discussion by phoebusryan** [pimcore/pimcore#17131] [Q&A / Support] [answered]
I recently updated our pimcore from 5.8.9 nto 6.9.6 and tried to update it again to version 10. I used this tutorial (https://pimcore.com/docs/platform/Pimcore/Installation_and_Upgrade/Updating_Pimcore/V6_to_V10/)

I think everything was fine until I verified the installation with:
`./bin/console doctrine:migrations:up-to-date --prefix=Pimcore\\Bundle\\CoreBundle`

There was an error 'out-of-date' .. so I tried to do the migration manually:
`./bin/console doctrine:migrations:migrate --prefix=Pimcore\\Bundle\\CoreBundle.`

After a few seconds I got this error:

![pim_error_4](https://github.com/pimcore/pimcore/assets/4052863/ee42b9d2-3a9a-43bc-b457-1ead6471c887)

What can I do?
`mysql --version`

 ...gives me:
mysql Ver 15.1 Distrib 10.6.16-MariaDB, for debian-linux-gnu (x86_64).

It's also not possible to view pimcore in browser. There is only a 404 default-message. Maybe an htaccess issue? I updated this file.

**Comment by blankse** [comment]
@phoebusryan Have your MariaDB the right settings? Execute following command:
`./bin/console pimcore:system:requirements:check`

**Reply by phoebusryan** [reply]
Thanks for your response. This command gives me 'OK' in every line oft the MySQL-Section and e few Warnings in the PHP-Section

**Reply by blankse** [reply, marked as answer]
@phoebusryan Can you try to execute following SQL command before you migrate?
```
OPTIMIZE TABLE `documents_page`;
```

**Reply by phoebusryan** [reply]
Output was:
```
Table does not support optimize, doing recreate + analyse instead
OK
```

Then I retried
`./bin/console doctrine:migrations:migrate --prefix=Pimcore\\Bundle\\CoreBundle.`

And I think it worked.

Verification with:
`./bin/console doctrine:migrations:up-to-date --prefix=Pimcore\\Bundle\\CoreBundle`
was successfull.

So... thanks a lot!

It's still not accessible via Browser. Do you have an idea what's the issue here? It worked before the update.

**Reply by blankse** [reply]
Which webserver do you use? Apache or Ngnix? For Apache you need to create a .htaccess file in public folder (or use Virtual Hosts):
https://pimcore.com/docs/pimcore/10.6/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/Apache_Configuration.html

For Ngnix see here: https://pimcore.com/docs/pimcore/10.6/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/Nginx_Configuration.html

**Reply by phoebusryan** [reply]
I use Apache. I my public folder is a  .htaccess file as described here:
https://pimcore.com/docs/platform/Pimcore/Installation_and_Upgrade/Updating_Pimcore/V6_to_V10/#update-your-htaccess-or-your-nginx-configuration

I downloaded this file:
https://pimcore.com/docs/platform/Pimcore/Installation_and_Upgrade/System_Setup_and_Hosting/Apache_Configuration/

Now I just replaced it with your .htaccess file and still got just an 404.

**Reply by blankse** [reply]
In Pimcore 6.x the document root is web/, in Pimcore 10.x it is changed to public/. Maybe you changed not the document root in your webserver? So the webserver is looking for the web/ folder?

**Reply by phoebusryan** [reply]
I had that thought too
`/etc/apache2/sites-available/pimprod.conf`
links to
`/var/www/html/pimcore/pimprod/redirect`

And the redirect-Folder contains another .htaccess File (don't know why. Isn't this by default?) Filecontains:
```
RewriteEngine ON
RewriteConf %{HTTPS} !=on
RewriteRule ^/?(.*) https://%{SERVER_NAME}/$1 [R,L]
```

But it must surely be something along these lines. Where is it defined by default that it points to the web directory?

[EDIT ]
Found another configfile and there was a 'web'. I replaced it with 'public' and now IO get a error 500. Better than nothing. :)
I will debug it now.

[EDIT 2]
It's a fatal error:
`Call to undefined method Pimcore\\Bootstrap:Bootstrap() in public/index.php on line 21`

**Reply by blankse** [reply]
@phoebusryan Should be `\Pimcore\Bootstrap::bootstrap();`. See:
https://github.com/pimcore/skeleton/blob/10.2/public/index.php

**Reply by phoebusryan** [reply]
Interesting. Found nothing about that in the Upgrade-Tutorial. I just renamed my old app.php to index.php

... of course it worked and I've got output now. 

Now "good" output" but still.... :)

![symfony_error](https://github.com/pimcore/pimcore/assets/4052863/87c478ca-351a-40d1-acbd-06b380e74a39)

**Reply by blankse** [reply]
@phoebusryan The admin interface is working? Can you give the root document a proper controller? Not only `default`?

**Reply by phoebusryan** [reply]
frontend / backend hast the same issue. How can I change the root document controller? We only used pimcore for our productdata; we did not modify it so 'default' must be the initial default..? 

Just replaced the services.yaml with this one:
`https://raw.githubusercontent.com/pimcore/skeleton/10.2/config/services.yaml` and now i've got a different message

![symfony_error2](https://github.com/pimcore/pimcore/assets/4052863/cfdbd21f-93ad-46e2-8b54-90368cce89ec)

**Reply by blankse** [reply]
You should also create:
https://github.com/pimcore/skeleton/blob/10.2/src/Controller/DefaultController.php

**Reply by phoebusryan** [reply]
And again.. interesting. Found nothing about that in the tutorial. I copied this file and now I got the error
`Compile Error: Cannot declare class App\Controller\DefaultController, because the name is already in use`

I think this happens, because I did:
`Add "AppBundle\\": ["src/"], to the autoload block, if you would like to keep using AppBundle instead of App.`
... so the file would be loaded twice? [ah, nope, tried without that and than the Controller wasn't found]



It is very tedious how slowly I am progressing. Almost to the point of despair. Thought with an unmodified Pimcore and instructions like the official tutorials it would go fast :O

In any case, thank you very much for your help. Without you it would take infinitely longer!

**Reply by blankse** [reply]
@phoebusryan If you want keep AppBundle, than you musst change the DefaultController Namespace back to AppBundle and change the services.yaml back with "AppBundle" Namespaces for Controller/Commands/etc...

**Reply by phoebusryan** [reply]
@blankse I think i need the AppBundle..

I changed in services.yaml all App\xxxx to AppBundle\xxxx and also changed the namespace in DefaultController.php to AppBundle\Controller

Now i've got this again:
`The controller for URI "/" is not callable: Controller "default" does neither exist as service nor as class.`

**Reply by blankse** [reply]
Try to set following setting in the config/config.yaml
```
pimcore:
    documents:
        default_controller: AppBundle\Controller\DefaultController::defaultAction
```

**Reply by phoebusryan** [reply]
Unfortunaly that didn't work. My config.yaml looks like this now

```
imports:
    - { resource: 'local/' }

pimcore:
    documents:
        default_controller: AppBundle\Controller\DefaultController::defaultAction
```

**Comment by blankse** [comment]
@phoebusryan I think you have "default" as Controller in Documents. This is not allowed. It musst be a php class or a symfony service.

Can you execute the SQL ```SELECT controller FROM `documents_page`;``` to see the controllers.

You can remove the 'default' controller, than it should use the default controller in the config.yaml setting.
```UPDATE `documents_page` SET controller = null WHERE controller = 'default';```

**Reply by phoebusryan** [reply]
You're right again. This query returns two Rows. One is called "Default" and the other is called "default".

Your Query updated both Rows. I did than `composer update` again and it's working now. I can login and see my data. Damn, you're such a genius! Thanks a lot.

**Reply by phoebusryan** [reply]
@blankse What's the best way to move this installation now to a docker container? I hope updating would be easier with it.

**Reply by blankse** [reply]
@phoebusryan You can use https://github.com/pimcore/docker

**Reply by phoebusryan** [reply]
@blankse Funny thing... The frontend never worked and I never tested it. It doesn't matter because we only use the backend... :)

I get the error:
"Compile Error: Cannot declare class AppBundle\Controller\DefaultController, because the name is already in use"

I think that's because in composer.json I've got App\\ and AppBundle\\ ... but that's exactly what I found in the Docs:
https://pimcore.com/docs/platform/Pimcore/Installation_and_Upgrade/Updating_Pimcore/V6_to_V10/#cleanup-your-composerjson

Do you havee an idea to fix this?

**Reply by blankse** [reply]
@phoebusryan The error says that you have defined 2 DefaultController PHP classes. Have a look at your PHP files in the src\Controller folder. Are there 2 controllers with this name?

**Reply by phoebusryan** [reply]
Nope, there is only one File with that name. And when I rename the classname of this file, the errormessage also changes. It seems that the file will be loaded twice.

**Reply by blankse** [reply]
@phoebusryan Okay, then you should use either App or AppBundle and not both :)

When you change the setting you have to execute `composer dump-autoload`
See: https://getcomposer.org/doc/03-cli.md#dump-autoload-dumpautoload
